python/air/dialects/_air_ops_ext.py

```diff
@@ -227,6 +229,18 @@ def __init__(
 dma_memcpy_nd = DmaMemcpyNd
 
 
+def air_module(module_function):
+    @functools.wraps(module_function)
+    def air_module_wrapper(*args, **kwargs):
+        with Context() as ctx, Location.unknown():
```

Comment:
@jgmelber I am unsure if it makes sense to encapsulate the context within this decorator or not. In the spirit of removing boilerplate, it did seem to make sense for the programming examples. However, this then causes problems with some of the structure of ```tests/xrt/01``` though ```tests/xrt/01``` which perform additional logic/transforms within the context (if I am reading those tests correctly).

It's possible I could make an additional decorator or parameterize this decorator to take as an argument an additional optional parameter, a function pointer which could be applied to the module within the context.
What do you think?

Reply:
@erwei-xilinx @fifield would have the most informed opinion on this

Reply:
I figured out that I could get the context manually from ```module.context``` so I went ahead and moved the decorator to all the tests I found that seemed like they could use it. The only real downside I see is that it's hard to instantiate some typing parameters (the ```dtype``` argument in some of the ```test/xrt``` tests I modified)... but that parameter was hardcoded in all the tests so it doesn't seem like it's a current issue. I think there are ways around this issue, for example, by passing a ```dtype_representor``` (maybe a ```str``` or an ```int``` instead of a ```dtype``` directly) as an argument which could then get converted into the right type within the function wrapped by ```@module_builder```

However, I'd still welcome any feedback on this overall!

Reply:
Thank you Erika for your thorough analysis. It seems that your `module_builder` results in a cleaner syntax overall so I think we can proceed with it.